gaj23/bigbirds
# Gaj/refactor/add link

#### Purpose: 
> Check all that apply.

### What does this PR do? 
> Clearly and concisely describe changes made. Include screenshots as needed.

This refactor make "Guess the Bird" clickable and redirects the user to the details page. 

Closes Issues #17 #22 

Additionally, it fixes the home route from `/home` to just `/`.

### Problems Encountered & Resulting Solutions  
> State any problems encountered & include screenshots as needed. 
> Describe your code changes in detail for reviewers.

When trying to provide randomly generated data, I found some difficulty in applying the `useEffect` hook. I think part of this was due to my continued learning about React-Hooks. I was simultaneously attempting to assign a random bird with a typical function that I called inside of the component render, while also trying to apply the `useEffect` hook upon changes. This resulted in a "double update" and confusion when a user clicked on the "guess that bird" and we redirected to a bird that wasn't the bird they were told to guess. 

Moving all this logic inside of `useEffect` helped me understand what data I was pulling on. Moving any logic out of the rendering process and just having a variable called upon rather than a function also created a clearer picture of what data I was using. 

### Where should the reviewer start?
> What file(s) and line(s) are where changes are made?

Please see `Highlight.js` for a majority of the changes. 

More image data points were added inside of the `data` folder.

### How should this be tested?
> What steps need to be taken to confirm functionality? 

Please open the [deployed link](https://bigbirds.vercel.app/). Please click the "Guess that Bird" and check if you're redirected to the "learn more about" page. 

If any errors occur in the console or on screen, please submit a new issue [here](https://github.com/gaj23/bigbirds/issues)


### Steps To Take
> Any questions and possible steps moving forward.

I think my next step would be refactoring styling with a focus on 
- SASS
- desktop view (media queries added)


#### Checklist:
